sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -2565,10 +2565,10 @@ pi_result piMemBufferCreate(pi_context Context, pi_mem_flags Flags, size_t Size,
                             const pi_mem_properties *properties) {
 
   // TODO: implement read-only, write-only
-  if ((Flags & PI_MEM_FLAGS_ACCESS_RW) == 0) {
-    die("piMemBufferCreate: Level-Zero implements only read-write buffer,"
-        "no read-only or write-only yet.");
-  }
+  // if ((Flags & PI_MEM_FLAGS_ACCESS_RW) == 0) {
+  //   die("piMemBufferCreate: Level-Zero implements only read-write buffer,"
+  //       "no read-only or write-only yet.");
+  // }
```

Comment:
I want to talk about this section before marking this review ready.  Presently in memory_manager.cpp (see below) we only use the PI_MEM_FLAG_ACCESS_RW flag. So memory is always marked for read and write.  But for sRGB images, our OpenCL implementation will only accept images with memory marked as read only.     The place where that flag is set is used by both buffers and images.  

Here in the LevelZero PI, we see that `die` is called if the flag is not `ACCESS_RW`.  Removing this check has no consequence that I can see, and everything works.  I don't see an easy way in memory_manager.cpp to screen for backend, nor for buffer vs. image.  So am wanting opinions on just removing this check. Here I've commented it out.

Reply:
I think removing this die is fine since there is no read-only or write-only support in Level-Zero. It's still a good idea to check for expected `Flags`, now PI_MEM_FLAG_ACCESS_RW & PI_MEM_FLAG_ACCESS_RO

Reply:
So, in this patch image is created (clCreateImage) with CL_MEM_READ_ONLY flag if const user data is passed to initialize the image. Which means:
1. user is required to pass host data ptr for this scenario -> this is not mentioned in doc
2.  it must be const -> this is mentioned in doc
Same logic will be applied to buffers after this change. 

But there might be some problems with that. If user provided host data for initialization then it doesn't necessarily mean that data is copied back -> I believe there is a separate flag for that.

I.e. we can have the following scenario:
1. Const host data is provided for image initialization
2. But accessor is read_write
With this patch this scenario will fail because image/buffer will be created with READ_ONLY flag.

Reply:
> Probably we should create read-only/write-only images/buffers based on a access mode provided to an accessor

That is how SYCL does things generally. The problem here is that OpenCL Images also have access flags on _construction_, and, at least for sRGB, these must be set correctly or you'll get a `CL_IMAGE_FORMAT_NOT_SUPPORTED`.

Maybe if the user provides `const` data we should not allow the creation of writing accessors?

Reply:
Yes, I understand that this is about flags for OpenCL image construction.
I was trying to say that most likely OpenCL image construction happens only when accessor is created and not during sycl::image constructor. We have AllocaCommand in the scheduler which calls memory_manager::alloca when it's enqueued.

> Maybe if the user provides const data we should not allow the creation of writing accessors?

There might be problem with that, I was trying to describe it above.
If I am not mistaken, it is completely legal to have const data but write accessor **if data is not copied back**.
const data is used just to initialize image. But in the kernel we can write to image (writing accessor).

Reply:
Presently (before this PR) one could pass in host const data when creating an image or buffer, and that alone would not trigger any warnings or errors. And one can, I presume, use any accessor. So the issue, if there is one, would be encountered on copyback.  But, assuming this is an accurate description of the issue, that means it also exists before this PR and this PR doesn't change or worsen the problem.   

So maybe I should remove the commented lines, or update the check to include RO, and file a separate ticket on this issue? I see TODO: flags there, so perhaps there is already a tracker for it.